diff -ru emacs-30.1.orig/lisp/info.el emacs-30.1/lisp/info.el
--- emacs-30.1.orig/lisp/info.el	2025-08-12 20:05:43.113072679 +0200
+++ emacs-30.1/lisp/info.el	2025-08-12 20:20:15.483124982 +0200
@@ -480,6 +480,7 @@
       '( (".gz"       . "gunzip")
 	 (".z"        . "gunzip")
 	 (".bz2"      . ("bzip2" "-dc"))
+	 (".lzma"     . ("lzma" "-dc"))
 	 (".inz"      . "gunzip")
 	 (".igz"      . "gunzip")
 	 (".info.Z"   . "gunzip")
@@ -498,6 +499,7 @@
        (".info.gz"   . "gunzip")
        (".info.z"    . "gunzip")
        (".info.bz2"  . ("bzip2" "-dc"))
+       (".info.lzma" . ("lzma" "-dc"))
        (".info.xz"   . "unxz")
        (".info.zst"  . ("zstd" "-dc"))
        (".info.lz"   . ("lzip" "-dc"))
@@ -506,6 +508,7 @@
        ("-info.Y"    . "unyabba")
        ("-info.gz"   . "gunzip")
        ("-info.bz2"  . ("bzip2" "-dc"))
+       ("-info.lzma" . ("lzma" "-dc"))
        ("-info.z"    . "gunzip")
        ("-info.xz"   . "unxz")
        ("-info.zst"  . ("zstd" "-dc"))
@@ -516,6 +519,7 @@
        ("/index.gz"  . "gunzip")
        ("/index.z"   . "gunzip")
        ("/index.bz2" . ("bzip2" "-dc"))
+       ("/index.lzma" . ("lzma" "-dc"))
        ("/index.xz"  . "unxz")
        ("/index.zst" . ("zstd" "-dc"))
        ("/index.lz"  . ("lzip" "-dc"))
@@ -525,6 +529,7 @@
        (".gz"        . "gunzip")
        (".z"         . "gunzip")
        (".bz2"       . ("bzip2" "-dc"))
+       (".lzma"       . ("lzma" "-dc"))
        (".xz"        . "unxz")
        (".zst"       . ("zstd" "-dc"))
        (".lz"        . ("lzip" "-dc"))
