--- emacs-24.2/lisp/info.el.bak	2012-09-03 18:14:19.996758539 -0400
+++ emacs-24.2/lisp/info.el	2012-09-03 18:16:23.073195720 -0400
@@ -414,12 +414,14 @@
        (".info.gz"   . "gunzip")
        (".info.z"    . "gunzip")
        (".info.bz2"  . ("bzip2" "-dc"))
+	   (".info.lzma" . ("lzma" "-dc"))
        (".info.xz"   . "unxz")
        (".info"      . nil)
        ("-info.Z"    . "uncompress")
        ("-info.Y"    . "unyabba")
        ("-info.gz"   . "gunzip")
        ("-info.bz2"  . ("bzip2" "-dc"))
+	   ("-info.lzma" . ("lzma" "-dc"))
        ("-info.z"    . "gunzip")
        ("-info.xz"   . "unxz")
        ("-info"      . nil)
@@ -428,6 +430,7 @@
        ("/index.gz"  . "gunzip")
        ("/index.z"   . "gunzip")
        ("/index.bz2" . ("bzip2" "-dc"))
+	   ("/index.lzma". ("lzma" "-dc"))
        ("/index.xz"  . "unxz")
        ("/index"     . nil)
        (".Z"         . "uncompress")
@@ -435,6 +438,7 @@
        (".gz"        . "gunzip")
        (".z"         . "gunzip")
        (".bz2"       . ("bzip2" "-dc"))
+	   (".lzma"      . ("lzma" "-dc"))
        (".xz"        . "unxz")
        (""           . nil)))
   "List of file name suffixes and associated decoding commands.
